monkey/infection_monkey/master/automated_master.py

```diff
@@ -240,4 +245,8 @@ def _run_plugins(
         logger.info(f"Finished running {plugin_type}s")
 
     def cleanup(self):
-        pass
+        timestamp = time.time()
+        agent_shutdown_event = AgentShutdownEvent(
+            source=get_agent_id(), stop_time=timestamp, timestamp=timestamp
+        )
+        self._agent_event_queue.publish(agent_shutdown_event)
```

Comment:
This is not the master's responsibility. This belongs in `monkey.py`.

This should  replacing the `StateTelem(is_done=False...)`, though we're not quite ready to remove that telemetry yet.